hdf/src/hdf_priv.h

```diff
@@ -25,6 +25,10 @@
 /* The library always uses UNIXBUFIO */
 #define FILELIB UNIXBUFIO
 
+/* Common library headers */
+#include "hdf.h"
+#include "herr_priv.h"
+
```

Comment:
hdf.h brings in the H4_WORDS_BIGENDIAN constant through h4config.h. This must be done *before* we set DF_MT, below.